syntax = "proto3";

package chess;

service ChessEngine {
    rpc GetBestMove (GetBestMoveRequest) returns (GetBestMoveResponse);
}

message GetBestMoveRequest {
    string fen = 1;
    repeated string move_stack = 2;
    int32 depth = 3;
    int32 processes = 4;
}

message GetBestMoveResponse {
    string best_move = 1;
    float time_taken = 2;
    float eval = 3;
}
